### ISSUE TITLE
Non-deterministic runs

### ISSUE THREAD

**Issue by RishabGoel** [denisyarats/drq#10]
Hi,

I get a different train and eval curves every time I start the training even with the same seed. Is that supposed to happen even after setting all seeds i.e. random, numpy, torch(both cuda and non-cuda)? Did you observe a similar behaviour?

**Comment by ikostrikov** [comment]
Hi,

Do you experience this problem only when you run it on GPU or on CPU as well?

If it happens only on GPU, you need to add these lines:
https://github.com/ikostrikov/pytorch-a2c-ppo-acktr-gail/blob/master/main.py#L29

Let us know if adding these lines doesn't help, we will investigate the issue further.

**Comment by denisyarats** [comment]
Hi @RishabGoel,

Yes, I observed this behavior as well. There reason why it is happening is that because MuJoCo rendering via OpenGL is non deterministic -- occasionally you can see that some pixels are off by +-1 that they should be.

See this issue as well: https://github.com/deepmind/dm_control/issues/100.

**Comment by RishabGoel** [comment]
Hi @ikostrikov,

Thanks for a quick response. I have those lines in my code and I still get quite different runs.

Hi @denisyarats ,

Thanks for pointing to the issue! I am facing the same issue that the pixels are not the same. Did the suggestion in the issue help to solve the non-determinism problem? Or did you find a solution for this?

**Comment by denisyarats** [comment]
I haven't try their solution, neither I resolved the problem by other means. Would be curious to hear from you if you manage to make any progress here.

**Comment by RishabGoel** [comment]
sure @denisyarats! I will update you if I end up solving this issue. :)

Thanks @ikostrikov and @denisyarats for such a quick response
